fms/__main__.py

```diff
@@ -1,120 +1,121 @@
 
-def get_all_src_status():
-    get_src_controllers()
-def get_src_controllers(client: Client=None) -> List[str]:
+
+def delete_sensor(sensor_name, client=None):
     if client is None:
         client = Client(path=fms_happi_database)
-    ret = client.search(name='booch')
-    print(ret)
+    item = client.find_item(name=sensor_name)
+
+    if type(item) is FMSRaritanItem:
+        try:
+            parent_switch_item = client.find_item(name=item.parent_switch)
+            port = "port" + str(item.root_sensor_port)
+            sensor_list = getattr(parent_switch_item, port)
+            sensor_list = [sensor for sensor in sensor_list if sensor[0] != sensor_name]
+            print(f"deleted: {sensor_name} from: {sensor_list}")
+            setattr(parent_switch_item, port, sensor_list)
+            parent_switch_item.save()
+        except SearchError:
+            ...
+    client.remove_item(item)
 
-def add_src_controller():
-    controller_name = te.get_str("Enter controller name\n")
 
+def validate():
     client = Client(path=fms_happi_database)
-
-    item = client.create_item(item_cls=FMSSRCItem,
-        name=controller_name,
-        prefix="RTD:TEST:FMS"
-    )
-    item.save()
-
-def add_fms_sensor():
-    '''
-    container_type = None
-    root_sensor_port = None
-    eth_dist_last = None
-    root_sensor = None
-    sensor_name = te.get_str("Enter sensor name\n")
-    sensor_type = te.get_list_str(["Beckhoff", "Raritan"], "Enter sensor type Beckhoff/Raritan\n")
-
-    if sensor_type == "Raritan":
-        parent_switch = te.get_str("Enter Parent SRC Controller name:\n")
-        root_sensor_port = te.get_str("Enter Port Number if root[0-8]\n")
-        eth_dist_last = te.get_int("Enter Eth Distance from last sensor\n")
-        root_sensor = te.get_bool("Is this the first sensor connected to the switch? True/False\n")    
-        container_type = FMSRaritanItem
+    results = client.validate()
+    if len(results) == 0:
+        print("Success! Valid FMS Database.")
     else:
-        container_type = FMSBeckhoffItem
-    
+        print(f"These devices are malformed! {results}")
 
-    client = Client(path=fms_happi_database)
 
-    item = client.create_item(item_cls=container_type,
-        name=sensor_name,
-        prefix="RTD:TEST:FMS",
-        parent_switch=sensor_name,
-        root_sensor=root_sensor,
-        root_sensor_port=root_sensor_port,
-        eth_dist_last=eth_dist_last
-    )
-    '''
-    client = Client(path=fms_happi_database)
+def add_src_controller(controller_name=None, client=None):
+    add_sensor(controller_name, client)
+
+
+def add_fms_sensor(sensor_name=None, client=None):
+    add_sensor(sensor_name, client)
 
-    item = client.create_item(item_cls=OphydItem,
-        name="foob",
-        prefix="RTD:TEST:FMS",
+
+def launch_nalms():
+    command = "slam --topics FMS-alarms --bootstrap-servers 172.24.5.232:9094"
+    sys.subprocess(command, shell=True)
```

Comment:
Nitpick: Normal best practice is to use the `subprocess` module for this, maybe it doesn't matter much here

Reply:
imported subprocess